app/models/nic/base.rb

```diff
@@ -318,14 +318,14 @@ def not_required_interface
     def exclusive_primary_interface
       if host && primary?
         primaries = host.interfaces.select { |i| i.primary? && i != self }
```

Comment:
just a thought while we're here, we don't necessarily need to iterate over all and create a new array, we could do something like:
```suggestion
        duplicate = host.interfaces.any? { |i| i.primary? && i != self }
```
and then `error... if duplicate` (and same for provision)